.github/workflows/update-version.yml

```diff
@@ -0,0 +1,39 @@
+name: update-version
+
+# This is supposed to be "Reusable Workflow", and included in other workflow file(s).
+# This workflow should be triggered with "repository_dispatch" event with payload:
+# - github.event.client_payload.property_name, with value: "killbill-commons.version",
+#   "killbill-base-plugin.version", "killbill-platform.version", or "killbill-client.version".
+# - github.event.client_payload.property_value
+
+on:
+  workflow_call:
+
+jobs:
+  update_version:
+    name: Update ${{github.event.client_payload.property_name}} from Repository Dispatch
+    runs-on: ubuntu-22.04
+    if: github.event.client_payload.property_name != '' && 
+        github.event.client_payload.property_value != '' &&
+        contains(fromJSON('["killbill-commons.version", "killbill-base-plugin.version", "killbill-platform.version", "killbill-client.version"]'), github.event.client_payload.property_name)
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup git user
+        env:
+          BUILD_USER: ${{ secrets.BUILD_USER }}
+          BUILD_TOKEN: ${{ secrets.BUILD_TOKEN }}
+        run: |
+          git config --global user.email "contact@killbill.io"
+          git config --global user.name "Kill Bill core team"
+          git config --global url."https://${BUILD_USER}:${BUILD_TOKEN}@github.com/".insteadOf "git@github.com:"
+      # Cannot use Mudlet/xmlstarlet-action@master because always get "Invalid expression: //x:properties/x:${POM_PROPERTY_NAME}"
+      - name: Install XML Starlet
+        run: sudo apt-get install xmlstarlet
+      - name: Update ${{github.event.client_payload.property_name}} Property in pom.xml file
+        run: xmlstarlet ed --inplace -N x="http://maven.apache.org/POM/4.0.0" -u "//x:properties/x:${{github.event.client_payload.property_name}}" -v ${{github.event.client_payload.property_value}} pom.xml
```

Comment:
Does it really work for the multi-module repo - e.g `killbill-commons`; Don't we need to update the `pom.xml` from all the children directory, e.g https://github.com/killbill/killbill-commons/blob/master/automaton/pom.xml#L25 ?

If this is the case, why not using `mvn versions:set -DnewVersion=${{github.event.client_payload.property_value}}` ?

Reply:
The purpose of this update-version.yml is to update and commit this `/project/properties/<any>` in pom.xml:

```xml
<project>
  <properties>
    <!-- this will be updated -->
    <killbill-commons.version>1.2.3</killbill-commons.version>
  </properties>
</project>
```

meanwhile, AFAIK `mvn versions:set` to actually update its/dependencies version. 

Nevertheless, now I think this `update-version.yml` would never used outside of `killbill-oss-parent` repository :thinking: . Should I just remove this and put in `killbill-oss-parent` instead?